File: python/pumapy/utilities/timer.py
import time


class Timer:
    current = 0

    def __init__(self):
        self.current = time.perf_counter()

    def start(self):
        self.current = time.perf_counter()

    def reset(self):
        self.current = time.perf_counter()

    def elapsed(self):
        return time.perf_counter() - self.current

    def print_elapsed(self, msg=""):
        print(msg, (time.perf_counter() - self.current))

File: cpp/src/utilities/timer/timer.cpp
#include "timer.h"


puma::Timer::Timer()
{
    reset();
    myTime = 0;
}


bool puma::Timer::reset(){
    struct timeval tim11;
    gettimeofday(&tim11, NULL);
    currentTime=(tim11.tv_sec + (tim11.tv_usec / 1000000.0)) * 1000.0;
    return true;
}

double puma::Timer::elapsed(){
    struct timeval tim11;
    gettimeofday(&tim11, NULL);
    double newTime=(tim11.tv_sec + (tim11.tv_usec / 1000000.0)) * 1000.0;

    return (newTime-currentTime)/1000.0;

}

bool puma::Timer::start() {
    reset();
    return true;
}

bool puma::Timer::stop() {
    myTime += elapsed();
    return true;
}

